runtime/lua/vim/path.lua

```diff
@@ -0,0 +1,145 @@
+local uv = vim.loop
+
+local Path = {}
+
+function Path:__tostring()
+  return self.path
+end
+
+function Path:exists()
+  return uv.fs_realpath(self.path) and true
+end
+
+function Path:expanduser()
+  if self.path:match '^~' then
```

Comment:
Something like `~Documents` would expand, perhaps checking for an `/` or the end of the string after.

Reply:
I think this should be good now